paddle/fluid/framework/new_executor/interpreter/interpreter_util.cc

```diff
@@ -292,7 +297,8 @@ std::tuple<VariableValueMap, VariableIdMap> BuildVariableMap(
     VariableScope* var_scope,
     Scope* local_scope,
     bool allow_var_not_in_program = false,
-    bool allow_var_not_in_scope = false) {
+    bool allow_var_not_in_scope = false,
+    bool used_for_control_flow_op = false) {
```

Comment:
It seems that used_for_control_flow_op has the same behavior as allow_var_not_in_scope. Maybe you can just simply assign the value of used_for_control_flow_op to allow_var_not_in_scope, and It is unnecessary to add another control parameter.

Reply:
`used_for_control_flow_op` and `allow_var_not_in_scope ` work differently. 
`used_for_control_flow_op` work for all ops in control flow block.
`allow_var_not_in_scope` only work for the 'registered' op.
So if we registered `conditional_block_op` as `allow_var_not_in_scope`, mostly it only works for `conditional_block_op` in global block. The normal op in control flow block may still have inputs or outputs not occurred in `block.AllVars()`.  This doesn't solve the problem.

Reply:
It is just a difference about applicable OP, the code behavior is still all the same. You can simply change the assignment code of allow_var_not_in_scope into 
```C++
bool allow_var_not_in_scope = used_for_control_flow_op || ops_with_var_not_in_scope.count(op_type);
```
and the interface of BuildVariableMap does not need to be changed.